orion/primitives/timeseries_anomalies.py

```diff
@@ -39,6 +42,185 @@ def regression_errors(y, y_hat, smoothing_window=0.01, smooth=True):
+
+    The computed error is calculated as the difference between predicted
+    and expected values with a rolling smoothing factor.
+
+    Args:
+        y (ndarray):
+            Ground truth.
+        y_hat (ndarray):
+            Predicted values.
+
+    Returns:
+        ndarray:
+            An array of smoothed point-wise error.
+    """
+    return abs(y - y_hat)
+
+
+def _area_error(y, y_hat, score_window=10):
+    """Compute area error between predicted and expected values.
+
+    The computed error is calculated as the area difference between predicted
+    and expected values with a smoothing factor.
+
+    Args:
+        y (ndarray):
+            Ground truth.
+        y_hat (ndarray):
+            Predicted values.
+        score_window (int):
+            Optional. Size of the window over which the scores are calculated.
+            If not given, 10 is used.
+
+    Returns:
+        ndarray:
+            An array of area error.
+    """
+    smooth_y = pd.Series(y).rolling(
+        score_window, center=True, min_periods=score_window // 2).apply(integrate.trapz)
+    smooth_y_hat = pd.Series(y_hat).rolling(
+        score_window, center=True, min_periods=score_window // 2).apply(integrate.trapz)
+
+    errors = abs(smooth_y - smooth_y_hat)
+
+    return errors
+
+
+def _dtw_error(y, y_hat, score_window=10):
+    """Compute dtw error between predicted and expected values.
+
+    The computed error is calculated as the dynamic time warping distance
+    between predicted and expected values with a smoothing factor.
+
+    Args:
+        y (ndarray):
+            Ground truth.
+        y_hat (ndarray):
+            Predicted values.
+        score_window (int):
+            Optional. Size of the window over which the scores are calculated.
+            If not given, 10 is used.
+
+    Returns:
+        ndarray:
+            An array of dtw error.
+    """
+    length_dtw = (score_window // 2) * 2 + 1
+    half_length_dtw = length_dtw // 2
+
+    # add padding
+    y_pad = np.pad(y, (half_length_dtw, half_length_dtw),
+                   'constant', constant_values=(0, 0))
+    y_hat_pad = np.pad(y_hat, (half_length_dtw, half_length_dtw),
+                       'constant', constant_values=(0, 0))
+
+    i = 0
+    similarity_dtw = list()
+    while i < len(y) - length_dtw:
+        true_data = np.zeros((length_dtw, 2))
+        true_data[:, 0] = np.arange(length_dtw)
+        true_data[:, 1] = y_pad[i:i + length_dtw]
+
+        pred_data = np.zeros((length_dtw, 2))
+        pred_data[:, 0] = np.arange(length_dtw)
+        pred_data[:, 1] = y_hat_pad[i:i + length_dtw]
+
+        dtw, _ = sm.dtw(true_data, pred_data)
+        similarity_dtw.append(dtw)
+        i += 1
+
+    errors = ([0] * half_length_dtw + similarity_dtw +
+              [0] * (len(y) - len(similarity_dtw) - half_length_dtw))
+
+    return errors
+
+
+def reconstruction_errors(y, y_hat, step_size=1, score_window=10, smoothing_window=0.01,
```

Comment:
I just feel like timeseries_anomalies.py contains too many things.

We may discuss how to separate it into 2-3 sub-files.

Reply:
I am thinking maybe we can do:
* timeseries_errors -- with prediction based / reconstruction based error computation
* timeseries_anomalies -- for locating anomalies based on errors

would that help?

Reply:
Sure. Let do this way.